prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -47,10 +48,17 @@ def _discover(host: str) -> bool:
             except (SessionCreationError, InvalidCredentialsError):
                 # redfish available, wrong credentials or unable to create a session
                 result = True
+            except Exception:  # pylint: disable=W0718
+                # mark redfish not available in any other generic exception
+                result = False
             else:
                 # redfish available, login succeeded
                 result = True
-                redfish_obj.logout()
+            finally:
+                # make sure we logout in the end
+                if redfish_obj:
+                    redfish_obj.logout()
+                    logger.debug("(discover) Logged out from redfish service ...")
```

Comment:
Checking briefly the [redfish source code](https://github.com/DMTF/python-redfish-library/blob/main/src/redfish/rest/v1.py#L502), the client is also context manager, so why using it?

Reply:
It's a bit convoluted now, the `RedfishHelper` wraps `redfish_client`, so we won't directly use it's context manager, plus it will suffer the same issue as described in the description